File: backend/charges.py
def calculate_fare(base_fare, cost_per_minute, cost_per_mile, booking_fee, ride_duration_minutes, ride_distance_miles):
    fare = (
        base_fare +
        cost_per_minute * ride_duration_minutes +
        cost_per_mile * ride_distance_miles +
        booking_fee
    )
    return fare

# Example usage
base_fare = 5  # Base fare in currency
cost_per_minute = 50  # Cost per minute in currency
cost_per_mile = 500  # Cost per mile in currency
booking_fee = 1000  # Booking fee in currency
ride_duration_minutes = 30  # Ride duration in minutes
ride_distance_miles = 5  # Ride distance in miles

fare = calculate_fare(base_fare, cost_per_minute, cost_per_mile, booking_fee, ride_duration_minutes, ride_distance_miles)
print(f"Your Fare: {fare}")

File: backend/charges.js
export defualt function calculateFare(
  baseFare,
  costPerMinute,
  costPerMile,
  bookingFee,
  rideDurationMinutes,
  rideDistanceMiles
) {
  const fare =
    baseFare +
    costPerMinute * rideDurationMinutes +
    costPerMile * rideDistanceMiles +
    bookingFee;
  return fare;
}

// Example usage
const baseFare = 5; // Base fare in currency
const costPerMinute = 50; // Cost per minute in currency
const costPerMile = 500; // Cost per mile in currency
const bookingFee = 1000; // Booking fee in currency
const rideDurationMinutes = 30; // Ride duration in minutes
const rideDistanceMiles = 5; // Ride distance in miles

const fare = calculateFare(
  baseFare,
  costPerMinute,
  costPerMile,
  bookingFee,
  rideDurationMinutes,
  rideDistanceMiles
);
console.log(`Your Fare: ${fare}`);
